src/proxy/topic/ReliableTopicProxy.ts

```diff
@@ -150,6 +157,63 @@ export class ReliableTopicProxy<E> extends BaseProxy implements ITopic<E> {
         }
     }
 
+    publishAll(messages: any[]): Promise<void> {
+        const capacity = messages.length;
+        const reliableTopicMessages: Array<ReliableTopicMessage> = Array<ReliableTopicMessage>(capacity);
+
+        assertNotNull(messages);
+        for (const message in messages) {
+            assertNotNull(message);
+        }
+
+        try {
+            for (const message of messages) {
+                const reliableTopicMessage = new ReliableTopicMessage();
+                reliableTopicMessage.payload = this.serializationService.toData(message);
+                reliableTopicMessage.publishTime = Long.fromNumber(Date.now());
+                reliableTopicMessages.push(reliableTopicMessage);
+            }
+            switch (this.overloadPolicy) {
+                case TopicOverloadPolicy.ERROR:
+                    this.addMessagesOrFail(reliableTopicMessages).then(() => {
+                        return;
+                    });
+                    break;
+                case TopicOverloadPolicy.DISCARD_NEWEST:
+                    this.ringbuffer.addAll(reliableTopicMessages, OverflowPolicy.OVERWRITE).then(() => {
+                        return;
+                    });
+                    break;
+                case TopicOverloadPolicy.DISCARD_OLDEST:
+                    this.ringbuffer.addAll(reliableTopicMessages, OverflowPolicy.FAIL).then(() => {
+                        return;
+                    });
+                    break;
+                case TopicOverloadPolicy.BLOCK:
+                    this.addAndBlock(reliableTopicMessages, 100).then(() => {
+                        return;
+                    });
+                    break;
+                default:
+                    throw new IllegalArgumentError('Unknown overload policy');
+            }
+        } catch (e) {
+            if (e instanceof SchemaNotReplicatedError) {
+                return this.registerSchema(e.schema, e.clazz).then(() => this.publishAll(messages));
+            }
+            throw new HazelcastError(e + 'Failed to publish these messages' + messages + 'to topic' + this.name);
+        }
+        return Promise.resolve();
+    }
```

Comment:
```suggestion
    publishAll(messages: any[]): Promise<void> {
        const capacity = messages.length;
        const reliableTopicMessages: Array<ReliableTopicMessage> = Array<ReliableTopicMessage>(capacity);

        assertNotNull(messages);
        for (const message in messages) {
            assertNotNull(message);
        }

        try {
            for (const message of messages) {
                const reliableTopicMessage = new ReliableTopicMessage();
                reliableTopicMessage.payload = this.serializationService.toData(message);
                reliableTopicMessage.publishTime = Long.fromNumber(Date.now());
                reliableTopicMessages.push(reliableTopicMessage);
            }
            switch (this.overloadPolicy) {
                case TopicOverloadPolicy.ERROR:
                    return this.addMessagesOrFail(reliableTopicMessages);
                case TopicOverloadPolicy.DISCARD_NEWEST:
                    return this.ringbuffer.addAll(reliableTopicMessages, OverflowPolicy.OVERWRITE).then(() => {});
                case TopicOverloadPolicy.DISCARD_OLDEST:
                    return this.ringbuffer.addAll(reliableTopicMessages, OverflowPolicy.FAIL).then(() => {});
                case TopicOverloadPolicy.BLOCK:
                    return this.addAndBlock(reliableTopicMessages, 100);
                default:
                    return Promise.reject(new IllegalArgumentError('Unknown overload policy'));
            }
        } catch (e) {
            if (e instanceof SchemaNotReplicatedError) {
                return this.registerSchema(e.schema, e.clazz).then(() => this.publishAll(messages));
            }
            return Promise.reject(new HazelcastError(e + 'Failed to publish these messages' + messages + 'to topic' + this.name));
        }
    }
```

Handling of promises should be like this.

Reply:
Done.